### ISSUE TITLE
How's the progress?

### ISSUE THREAD

**Issue by Zheng-Changfu** [johndpope/VASA-1-hack#69]

**Comment by Zheng-Changfu** [comment]
Can reasoning be made based on photos and audio now

**Comment by johndpope** [comment]
im confident just need more training data - 
https://wandb.ai/snoozie/vasa-overfitting/runs/8mq2bmob?nw=nwusersnoozie


<img width="1024" height="256" alt="Image" src="https://github.com/user-attachments/assets/b93e863f-5f8e-4fed-aacd-81b98437a08f" />

emo can transfer identity well - 

the canonical code 
-> audio -> expression 

but i make it pass through first facial Action Units first so its kind of in a sweet spot to target expression
https://github.com/johndpope/VASA-1-hack/commit/139600ae18b198ae8ece34a0aafbf4466dc97d1b

the mouth voice sync - i introduce the sync loss from synchorama - 




   INFO     [DERIVED WARPS] Generating UV warps from  vasa_model.py:1590
2025-11-21 16:46:28
                    zdyn torch.Size([1, 50, 128]) + idt_embed
2025-11-21 16:46:28
                    torch.Size([1, 512, 4, 4])
2025-11-21 16:46:28
           INFO     [WARP DEBUG] idt_embed spatial:           vasa_model.py:1344
2025-11-21 16:46:28
                    torch.Size([1, 512, 4, 4]) -> repeated to
2025-11-21 16:46:28
                    torch.Size([50, 512, 4, 4]), zdyn:
2025-11-21 16:46:28
                    torch.Size([1, 50, 128])
2025-11-21 16:46:29
[16:46:29] INFO     [DERIVED WARPS] ✅ Generated warps shape: vasa_model.py:1601
2025-11-21 16:46:29
                    torch.Size([1, 50, 16, 64, 64, 3])
2025-11-21 16:46:29
           INFO     Using high-quality identity image for   vasa_trainer.py:1494
2025-11-21 16:46:29
                    source, shape: torch.Size([1, 3, 512,
2025-11-21 16:46:29
                    512]), range: [0.00, 1.00]
2025-11-21 16:46:29
           INFO     ✅ [GT THETA] Injected GT theta for     vasa_trainer.py:1539
2025-11-21 16:46:29
                    frame generation (isolating expression
2025-11-21 16:46:29
                    learning)
2025-11-21 16:46:29
           INFO       ✓ Diff from predicted: 3.834219       vasa_trainer.py:1540
2025-11-21 16:46:29
                    (should be >0)
2025-11-21 16:46:29
           INFO       ✓ Match with GT: 0.0000000000 (should vasa_trainer.py:1541
2025-11-21 16:46:29
                    be ~0)
2025-11-21 16:46:29
           INFO       📊 GT theta variance: 0.000242        vasa_trainer.py:1542
2025-11-21 16:46:29
                    (variance check in VA bridge uses 0.01
2025-11-21 16:46:29
                    threshold)
2025-11-21 16:46:29
           INFO     🔍 Motion parameter variance check     vasa_va_bridge.py:212
2025-11-21 16:46:29
                    (theta looks like PREDICTED):
2025-11-21 16:46:29
           INFO       theta variance across time: 0.000242 vasa_va_bridge.py:213
2025-11-21 16:46:29
           INFO       expression_embed variance across     vasa_va_bridge.py:214
2025-11-21 16:46:29
                    time: 0.007325
2025-11-21 16:46:29
           INFO       uv_warps variance across time:       vasa_va_bridge.py:215
2025-11-21 16:46:29
                    0.000165
2025-11-21 16:46:29
           INFO       uv_warps magnitude (mean abs):       vasa_va_bridge.py:216
2025-11-21 16:46:29
                    0.625307
2025-11-21 16:46:30
[16:46:30] INFO     Expression stats - Pred: mean=0.050,     vasa_losses.py:2133
2025-11-21 16:46:30
                    std=0.286 | Target: mean=0.043,
2025-11-21 16:46:30
                    std=0.438
2025-11-21 16:46:30
           INFO       Variance loss: 0.002295, Temporal      vasa_losses.py:2134
2025-11-21 16:46:30
                    loss: 0.000004
2025-11-21 16:46:30
           INFO     📊 Total reconstruction loss: 3.5523      vasa_losses.py:661
2025-11-21 16:46:30
           INFO     🔍 Expression diagnostics:                vasa_losses.py:669
2025-11-21 16:46:30
           INFO       Predicted std: 0.286465, Target std:    vasa_losses.py:670
2025-11-21 16:46:30
                    0.437583
2025-11-21 16:46:30
           INFO       Pred-Target diff: 0.217320              vasa_losses.py:671
2025-11-21 16:46:30
           INFO     [_compute_sync_loss_synchrorama] Entry:  vasa_losses.py:1749
2025-11-21 16:46:30
                    generated_frames.shape = torch.Size([1,
2025-11-21 16:46:30
                    50, 3, 512, 512])
2025-11-21 16:46:30
           INFO     ✅ [SynchroRaMa] Sync loss (offset MSE): vasa_losses.py:1770
2025-11-21 16:46:30
                    0.730099


BEWARE - 

i ignore the theta (head pose regression) prediction - and focus only on the mouth + lip sync 
that is until we stabilise this - 



N..b - you can use this - 58fps with implicit motion function code under the hood.
https://github.com/cbsjtu01/IMTalker/pull/5

**Comment by Zheng-Changfu** [comment]
Thank you for your answer. May I ask if it can be in real time

**Comment by johndpope** [comment]
I achieve 58fps with IMTalker (based off Microsoft IMF)
Http://www.github.com/johndpope/IMF

https://github.com/johndpope/IMTalker/tree/feat/tensor-rt

**Comment by fenghe12** [comment]
Hi, I wonder if you have access to this hf dataset repo https://huggingface.co/datasets/vigil1917/GazeGene. I may also use GazeGene dataset to train my talking face generation model.

**Comment by johndpope** [comment]
check out grok imagine - you can create synthetic datasets

**Comment by fenghe12** [comment]
Thank you for your suggestion. However, I actually need the GazeGene dataset to train a gaze-aware talking face generation model. It seems that Grok cannot generate facial data with accurate gaze annotations (i.e., precise yaw and pitch labels).

**Comment by johndpope** [comment]
this is a chrome extension ive been working on -  link may not work unless you're logged in to chrome store - https://chromewebstore.google.com/detail/scrya-%E2%80%94-ai-prompt-enhance/ilmddboidkofgicinlhocglmjddpogff - its pretty impressive using grok - 6 trillion model is in the works - and Musk has said it has to work in realtime.
